### PR TITLE
Add #refine support for Module

### PR DESCRIPTION
**Pull request by ssnickolay**
I thought about duplication for modules all the tests that are written for classes (through the [shared_examples](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples)), but for now it seems a little redundant.
From the point of view of refinement, the modules and classes are the same and there should be no problems.
Am I right or should I add more tests for modules? (maybe spy them in CRuby)

### CODE REVIEW

File: src/main/java/org/truffleruby/core/module/ModuleNodes.java [hunk 1]
```diff
@@ -2093,33 +2093,33 @@ protected void setMethodVisibility(DynamicObject module, Object name,
         @Child private CallBlockNode callBlockNode = CallBlockNode.create();
 
         @Specialization
-        protected DynamicObject refine(DynamicObject self, DynamicObject classToRefine, NotProvided block) {
+        protected DynamicObject refine(DynamicObject self, DynamicObject objectToRefine, NotProvided block) {
```
Comment (ssnickolay) [inline src/main/java/org/truffleruby/core/module/ModuleNodes.java:2096]: Not sure about the name `objectToRefine`. We should rename `classToRefine` but maybe just to `object`

Comment (eregon) [reply src/main/java/org/truffleruby/core/module/ModuleNodes.java:2096]: How about `moduleToRefine`? Classes are Modules (also holds for isRubyClass() implies isRubyModule()).

File: src/main/java/org/truffleruby/core/module/ModuleNodes.java [hunk 2]
```diff
@@ -2093,33 +2093,33 @@ protected void setMethodVisibility(DynamicObject module, Object name,
...
             throw new RaiseException(getContext(), coreExceptions().argumentError("no block given", this));
         }
 
-        @Specialization(guards = "!isRubyClass(classToRefine)")
-        protected DynamicObject refineNotClass(DynamicObject self, Object classToRefine, DynamicObject block) {
+        @Specialization(guards = "!(isRubyClass(objectToRefine) || isRubyModule(objectToRefine))")
```
Comment (eregon) [inline src/main/java/org/truffleruby/core/module/ModuleNodes.java:2100]: Can be simplified to `guards = "!isRubyModule(moduleToRefine)"`

**Comment by eregon** [comment]
> Am I right or should I add more tests for modules? (maybe spy them in CRuby)

It looks fine as in the PR, it seems overkill for now to run all refinement specs on both classes and modules.

File: src/main/java/org/truffleruby/core/module/ModuleNodes.java (hunk 1 above)
Comment (eregon) [reply src/main/java/org/truffleruby/core/module/ModuleNodes.java:2096]: The terminology is documented in https://github.com/oracle/truffleruby/blob/e276e66f38306842602833fb2357a9416580c889/src/main/java/org/truffleruby/core/module/ModuleFields.java#L80-L93

It's for an example like:
```ruby
module M # M is the namespace module
  R = refine C do # C is the refined module
  end
  # R is the refinement module
end
```

So I think `refinedModule` would be the best, and the `refinedClass` field in ModuleFields should be renamed to `refinedModule`.

**Comment by ssnickolay** [comment]
Thanks for the detailed review!
I agree with almost everything, just kept the word “class” in the code comments, because in a global sense Class != Module.

**Comment by ssnickolay** [comment]
//cc @palkan

**Pull request merged**